tokenizers/src/tokenizer/encoding.rs

```diff
@@ -155,11 +180,40 @@ impl Encoding {
             .for_each(func)
     }
 
+    /// Returns the range to target to retrieve something (word_id, offsets, ..) related to the
+    /// given sequence id
+    fn sequence_range(&self, sequence_id: usize) -> Range<usize> {
+        self.sequence_ranges
+            .get(&sequence_id)
+            .cloned()
+            .unwrap_or(0..self.len())
+    }
+
+    /// Returns the index of the sequence containing the given token
+    pub fn token_to_sequence(&self, token: usize) -> Option<usize> {
```

Comment:
Removing empty hashmap is empty mean one sequence over the whole lenght would really simplify a lot of the code too.